src/src/SessionStore.cpp

```diff
@@ -0,0 +1,206 @@
+    std::lock_guard<std::mutex> lock2(_tcpSessionsMutex);
+
+    auto udpNb = std::erase_if(_udpSessions, [&sessionid](auto &pair) {
+        return pair.second == sessionid;
+    }) != 0;
+    auto tcpNb = std::erase_if(_tcpSessions, [&sessionid](auto &pair) {
+        return pair.second == sessionid;
+    }) != 0;
+    return udpNb != 0 || tcpNb != 0;
+}
+
+polymorph::network::SessionId polymorph::network::SessionStore::sessionOf(asio::ip::udp::endpoint endpoint)
+{
+    try {
+        std::lock_guard<std::mutex> lock(_udpSessionsMutex);
+        return _udpSessions.at(endpoint);
+    } catch (std::out_of_range &e) {
+        throw exceptions::UnknownSessionException("Unknown session");
+    }
+}
+
+polymorph::network::SessionId polymorph::network::SessionStore::sessionOf(asio::ip::tcp::endpoint endpoint)
+{
+    try {
+        std::lock_guard<std::mutex> lock(_tcpSessionsMutex);
+        return _tcpSessions.at(endpoint);
+    } catch (std::out_of_range &e) {
+        throw exceptions::UnknownSessionException("Unknown session");
+    }
+}
+
+asio::ip::udp::endpoint polymorph::network::SessionStore::udpEndpointOf(polymorph::network::SessionId session)
+{
+    std::lock_guard<std::mutex> lock(_udpSessionsMutex);
+    auto it = std::find_if(_udpSessions.begin(), _udpSessions.end(), [session](auto &pair) {
+        return pair.second == session;
+    });
+
+    if (it == _udpSessions.end())
+        throw exceptions::UnknownSessionException("Unknown session");
+    return it->first;
+}
+
+std::vector<asio::ip::udp::endpoint>
+polymorph::network::SessionStore::udpEndpointsOfExcept(polymorph::network::SessionId except)
+{
+    std::lock_guard<std::mutex> lock(_udpSessionsMutex);
+    std::vector<asio::ip::udp::endpoint> sessions;
+
+    for (auto &session: _udpSessions)
+        if (session.second != except)
+            sessions.push_back(session.first);
+    return sessions;
+}
+
+std::vector<polymorph::network::SessionId> polymorph::network::SessionStore::allUdpSessions()
+{
+    std::lock_guard<std::mutex> lock(_udpSessionsMutex);
+    std::vector<SessionId> sessions;
+
+    for (auto &session: _udpSessions)
+        sessions.push_back(session.second);
+    return sessions;
+}
+
+polymorph::network::AuthorizationKey
+polymorph::network::SessionStore::generateUdpAuthorizationKey(polymorph::network::SessionId sessionId)
+{
+    std::lock_guard<std::mutex> lock(_udpSessionsAuthorizationKeysMutex);
+
+    try {
+        return _udpSessionsAuthorizationKeys.at(sessionId);
+    } catch (std::out_of_range &e) {
+        auto key = authorizationKey::generate();
+        _udpSessionsAuthorizationKeys.emplace(sessionId, key);
+        return key;
+    }
+}
+
+polymorph::network::AuthorizationKey
+polymorph::network::SessionStore::generateTcpAuthorizationKey(polymorph::network::SessionId sessionId)
+{
+    std::lock_guard<std::mutex> lock(_tcpSessionsAuthorizationKeysMutex);
+
+    try {
+        return _tcpSessionsAuthorizationKeys.at(sessionId);
```

Comment:
Using contains or any conditional presence will be better than an entire try catch block